mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -478,11 +496,46 @@ class ReduceMeanConverter final
   }
 };
 
+class QuantizeLinearConverter final
+    : public OpConversionPattern<migraphx::QuantizeLinearOp> {
+public:
+  using OpConversionPattern<migraphx::QuantizeLinearOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(migraphx::QuantizeLinearOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto input = op.getInput();
+    auto scale = op.getScale();
+    ShapedType inputType = input.getType().cast<ShapedType>();
+    auto elementType = inputType.getElementType();
+    Location loc = op->getLoc();
+
+    assert(!elementType.isInteger(8) &&
```

Comment:
Instead of an assert(), `notifyMatchFailure()`. Passing in bad input is not assertion-worthy unless such input is 100% our fault.

Reply:
It's probably 100% our fault (or migraphx fault) to ask a type beyond what they tell us to. But I agree assert is probably too strong for this type of issue.

Reply:
Done in 8aed021